tel2puml/find_unique_graphs/otel_ingestion/otel_data_source.py

```diff
@@ -27,3 +62,121 @@ def __next__(self) -> OTelEvent:
         :rtype: :class: `OTelEvent`
         """
         pass
+
+
+class JSONDataSource(OTELDataSource):
+    """Class to handle parsing JSON OTel data from a file or directory,
+    returning OTelEvent objects.
+    """
+
+    def __init__(self) -> None:
+        """Constructor method."""
+        super().__init__()
+        self.current_file_index = 0
+        self.current_parser: Iterator[OTelEvent] | None = None
+        self.dirpath = self.set_dirpath()
+        self.filepath = self.set_filepath()
+        if not self.dirpath and not self.filepath:
+            raise FileNotFoundError(
+                """No directory or files found. Please check yaml config."""
+            )
+        self.file_list = self.get_file_list()
+
+    def set_dirpath(self) -> str | None:
+        """Set the directory path.
+
+        :return: The directory path
+        :rtype: `str` | `None`
+        """
+        dirpath: str = self.yaml_config["data_sources"][f"{self.data_source}"][
+            "dirpath"
+        ]
+
+        if not dirpath:
+            return None
+        elif not os.path.isdir(dirpath):
+            raise ValueError(f"{dirpath} directory does not exist.")
+        return dirpath
+
+    def set_filepath(self) -> str | None:
+        """Set the filepath.
+
+        :return: The filepath
+        :rtype: `str` | `None`
+        """
+        filepath: str = self.yaml_config["data_sources"][
+            f"{self.data_source}"
+        ]["filepath"]
+
+        if not filepath:
+            return None
+        elif not filepath.endswith(f".{self.data_source}"):
+            raise ValueError(
+                f"File provided is not .{self.data_source} format."
+            )
+        elif not os.path.isfile(filepath):
+            raise ValueError(f"{filepath} does not exist.")
+        return filepath
+
+    def get_file_list(self) -> list[str]:
```

Comment:
Could we maybe make this recursively search through sub folders and get files as well?